go-controller/pkg/ovn/secondary_layer2_network_controller.go

```diff
@@ -299,6 +302,10 @@ func (oc *SecondaryLayer2NetworkController) Start(ctx context.Context) error {
 		klog.Infof("Starting controller for secondary network %s took %v", oc.GetNetworkName(), time.Since(start))
 	}()
 
+	if err := oc.ensureNetworkID(); err != nil {
+		return fmt.Errorf("failed ensuring network id at user defined layer2 network controller for network '%s': %w", oc.GetNetworkName(), err)
+	}
```

Comment:
why don't we embed this in the controller's init function ?

Reply:
We don't want to do this for localnet

Reply:
so ? There's a dedicated controller per topology and each has a different initialization function. Put it there in the controllers that need it - layer2 and layer3.